src/test/kotlin/com/sourcegraph/config/ConfigUtilTest.kt

```diff
@@ -0,0 +1,66 @@
+import com.google.gson.JsonParser
+import com.intellij.openapi.project.Project
+import com.sourcegraph.config.ConfigUtil
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Test
+import org.mockito.Mockito.mock
+
+class ConfigUtils {
+
+  private val mockProject = mock(Project::class.java)
+
+  @Test
+  fun testGetCustomConfiguration_addsAdditionalProperties() {
+    val input =
+        """
+    {
+      "cody.debug": true,
+      "cody.autocomplete.enabled": true
+    }
+    """
+    val result = ConfigUtil.getCustomConfiguration(mockProject, input)
+    val parsed = JsonParser.parseString(result).asJsonObject
+
+    assertEquals(
+        "indentation-based",
+        parsed
+            .get("cody")
+            .asJsonObject
+            .get("experimental")
+            .asJsonObject
+            .get("foldingRanges")
+            .asString)
```

Comment:
`withValue` adds the field as 
```
"cody": {
  "experimental": {
    "foldingRanges": "indentation-based"
  }
}
```
instead of 
```
"cody.experimental.foldingRanges": "indentation-based"
```

Is it a problem? 🤔

Reply:
IMHO It is not, those are equivalent in VSC config.